### ISSUE TITLE
Constraint Violation does not show violations in combination with micronaut-serde

### ISSUE THREAD

**Issue by debrutal** [micronaut-projects/micronaut-problem-json#176]
### Expected Behavior

As a consumer of an API i expect the Bean Validation to render a 400 with a Constration Violation and the list of violations.

Actual   :{"type":"https://zalando.github.io/problem/constraint-violation","title":"Constraint Violation","status":400}
Expected :{"type":"https://zalando.github.io/problem/constraint-violation","title":"Constraint Violation","status":400,"violations":[{"field":"greeting","message":"must not be blank"}]}

### Actual Behaviour

Actual   :{"type":"https://zalando.github.io/problem/constraint-violation","title":"Constraint Violation","status":400}

### Steps To Reproduce

- `git clone https://github.com/debrutal/serdeable.git`
- `mvn test`

### Environment Information

JDK 17.0.2

### Example Application

https://github.com/debrutal/serdeable

### Version

3.4.3

**Comment by debrutal** [comment]
Hint:
ThrowableProblemWithoutStacktrace does hide all relevant fields. Especially `violations` which is only desired in case of the ConstraintValidationException. 

Workarounds:
One way to mitigate this is to use Jackson. (not preferred)
Another way is to overwrite the ProblemErrorResponseProcessor, which will check for a ConstraintViolation Exception.

A well integrated solution preferred

**Comment by t-gebauer** [comment]
A possible Workaround by modifying the ProblemErrorResponseProcessor:
(Important parts marked with comments)

```java
package com.example;

import io.micronaut.context.annotation.Replaces;
import io.micronaut.core.annotation.NonNull;
import io.micronaut.http.HttpMethod;
import io.micronaut.http.MutableHttpResponse;
import io.micronaut.http.server.exceptions.response.ErrorContext;
import io.micronaut.problem.ProblemErrorResponseProcessor;
import io.micronaut.problem.conf.ProblemConfiguration;
import io.micronaut.problem.violations.ConstraintViolationThrowableProblem;
import io.micronaut.problem.violations.Violation;
import io.micronaut.serde.annotation.Serdeable;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;
import org.zalando.problem.Problem;
import org.zalando.problem.StatusType;
import org.zalando.problem.ThrowableProblem;

import java.net.URI;
import java.util.List;
import java.util.Map;

@Singleton
@Replaces(ProblemErrorResponseProcessor.class)
public class ConstraintViolationProblemResponseProcessor extends ProblemErrorResponseProcessor {

    private final boolean stackTraceConfig;

    @Inject
    public ConstraintViolationProblemResponseProcessor(ProblemConfiguration config) {
        super(config);
        stackTraceConfig = config.isStackTrace();
    }

    @Override
    @NonNull
    public MutableHttpResponse<Problem> processResponse(
        @NonNull ErrorContext errorContext,
        @NonNull MutableHttpResponse<?> baseResponse
    ) {
        if (errorContext.getRequest().getMethod() == HttpMethod.HEAD) {
            return (MutableHttpResponse<Problem>) baseResponse;
        }
        ThrowableProblem throwableProblem = errorContext.getRootCause()
                                                        .filter(t -> t instanceof ThrowableProblem)
                                                        .map(t -> (ThrowableProblem) t)
                                                        .orElseGet(() -> defaultProblem(errorContext, baseResponse.getStatus()));

        // 1. use custom wrapper only for ConstraintViolation
        if (throwableProblem instanceof ConstraintViolationThrowableProblem && !stackTraceConfig) {
            return baseResponse
                .contentType(APPLICATION_PROBLEM_JSON)
                .body(new ConstraintViolationThrowableProblemWrapper((ConstraintViolationThrowableProblem) throwableProblem));

        } else {
            return super.processResponse(errorContext, baseResponse);
        }
    }

    @Serdeable
    public static final class ConstraintViolationThrowableProblemWrapper implements Problem {

        final ConstraintViolationThrowableProblem problem;

        public ConstraintViolationThrowableProblemWrapper(ConstraintViolationThrowableProblem problem) {
            this.problem = problem;
        }

        @Override
        public URI getType() {
            return problem.getType();
        }

        @Override
        public String getTitle() {
            return problem.getTitle();
        }

        @Override
        public StatusType getStatus() {
            return problem.getStatus();
        }

        @Override
        public String getDetail() {
            return problem.getDetail();
        }

        @Override
        public URI getInstance() {
            return problem.getInstance();
        }

        @Override
        public Map<String, Object> getParameters() {
            return problem.getParameters();
        }

        // 2. additional violations field
        public List<Violation> getViolations() {
            return problem.getViolations();
        }
    }

}

**Comment by edeesis** [comment]
I actually think the problem here is because ThrowableProblemWithoutStacktrace is marked as Introspected (which is what serde uses), but it doesn't use Field Access Level.

Using custom Problems has this same issue, see my example project [here](https://github.com/edeesis/problem-serde-example).

**Comment by edeesis** [comment]
It would appear that using Field AccessKind still isn't working, the properties aren't being ignored by JsonIgnoreProperties either. This seems like a micronaut-serde issue as well: https://github.com/micronaut-projects/micronaut-serialization/issues/232

**Comment by edeesis** [comment]
So it looks like at one point, a mixin instead of a wrapper type was used in this project: https://github.com/micronaut-projects/micronaut-problem-json/pull/69. However, it was removed because Serde didn't support mixins (presumably at that time)

However, they now support mixins, so maybe that functionality can be added back?

**Comment by edeesis** [comment]
So what I ended up doing is, I created an interface with `@JsonIgnoreProperties("localizedMessage", "message", "stackTrace") and then added `problem.stack-trace: true` to my application.yml file.

Then I redeclared SerdeImport for Exceptional and DefaultProblem with this interface, added one for ConstraintViolationThrowableProblem, and also used this interface as a parent class to my own custom problem.

It's a little messy, but it works

**Comment by sdelamo** [comment]
I have released a patch release 2.4.1 and 2.5.1 which solves the issue for constraint violations. The issue with Custom problems and Serde is still present. It is being worked at: https://github.com/micronaut-projects/micronaut-problem-json/pull/196
